### DISCUSSION TITLE
Can you eexcute js with the read and write extension if so how??

### DISCUSSION THREAD

**Discussion by ghost** [3kh0/ext-remover#1406] [Help JOIN TITANIUM NETWORK TO SHARE PLEASE]
???

**Comment by eldorito-Xx01xX** [comment]
![image](https://github.com/user-attachments/assets/5e130cd5-8a4b-408f-99b1-41ec78c113b2)
it should be possible to run js stuff with this extension. its mv3, tho, so it has limited scripting capabilities (so no external URLs in the js.)
do you have a mv2 extension with the "tabs" API? as that would be better for doing code execution
Im using hapara highlights for code execution, and it's working fine.
![image](https://github.com/user-attachments/assets/dea949c8-1ab1-421f-b093-b9bbae458991)

**Reply by ghost** [reply]
ok thanks

**Reply by ghost** [reply]
No  ,  The only other run is gogudrain but it blocks all filesystem links, but thanks!

**Reply by eldorito-Xx01xX** [reply]
np

**Reply by duckman567** [reply]
hey, I have multiple extensions that are mv2 and have the tabs API, and I have filesystem: so I was wondering how I could get that js running thing you have, and I haven't dabbled much in running js on extensions, here is some links to some extensions that have mv2 and tabs api that I can install: 
https://chromewebstore.google.com/detail/strict-workflow/cgmnfnmlficgeijcalkgnnkigkefkbhd
https://chromewebstore.google.com/detail/outwrite-%E2%80%94-grammar-checke/jldbdlmljpigglecmeclifcdhgbjbakk https://chromewebstore.google.com/detail/select-and-speak-text-to/gfjopfpjmkcfgjpogepmdjmcnihfpokn
https://chromewebstore.google.com/detail/beeline-reader-pdf-viewer/pmeknoingfjncbdhempgnkdgojickcko
https://chromewebstore.google.com/detail/announcify/mmiolkcfamcbpoandjpnefiegkcpeoan

also, I have Snap&Read and Read&Write but Snap&Read isn't working, instead it does a doi.org resolve when I try to run js, and I use ltmeat flood so i'm not limited by goguardian but I just wanna run js

**Reply by Charlieis1nsan3** [reply]
wait hold up how do I do that?

**Reply by eldorito-Xx01xX** [reply]
late but it should work. you just input the extension url into rigtools, and run this piece of javascript to evaluate the potential to run code, and if it comes back with the top one, or you see the on tab update scripts at the top of the page, you should be good to go

alert(chrome?.tabs?.executeScript ? "mv2 extension, can exec with chrome.tabs" : chrome?.scripting?.executeScript ? "mv3 extension, limited exec with chrome.scripting.executeScript" : "cannot exec code on other tabs")

**Comment by Charlieis1nsan3** [comment]
Nooo I neverr got to use rig tools 😢😢😢😢
